From 877623a3bbdf45d16e82f804e7c086e030c65e73 Mon Sep 17 00:00:00 2001
From: jinghua <xigu@marvell.com>
Date: Wed, 24 Aug 2016 23:04:37 +0800
Subject: [PATCH 0479/1345] pinctrl: a3700: rename pinctrl settings to z1

commit  522b4cc1d59939e67df6dd6fe55441dc99db1e28 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

- Current pinctrl driver only supports Armada3700 Z1 chip.
- This patch renames pinctrl settings to Z1, to distinguish from A0.

Change-Id: Ibe9228d503f7b7a693952e1ee9aa477b2b4769fd
Signed-off-by: jinghua <xigu@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/32604
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Wilson Ding <dingwei@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../pinctrl/marvell,armada-3700-pinctrl.txt        |    6 +-
 arch/arm64/boot/dts/marvell/armada-37xx-rz.dtsi    |   12 ++++
 drivers/pinctrl/mvebu/pinctrl-armada-3700.c        |   60 ++++++++++----------
 3 files changed, 45 insertions(+), 33 deletions(-)

diff --git a/Documentation/devicetree/bindings/pinctrl/marvell,armada-3700-pinctrl.txt b/Documentation/devicetree/bindings/pinctrl/marvell,armada-3700-pinctrl.txt
index f833e38..22167e9 100644
--- a/Documentation/devicetree/bindings/pinctrl/marvell,armada-3700-pinctrl.txt
+++ b/Documentation/devicetree/bindings/pinctrl/marvell,armada-3700-pinctrl.txt
@@ -4,7 +4,7 @@ Please refer to marvell,mvebu-pinctrl.txt in this directory for common binding
 part and usage.
 
 Required properties:
-- compatible: "marvell,armada-3700-nb-pinctrl" or "marvell,armada-3700-sb-pinctrl"
+- compatible: "marvell,armada-3700-z1-nb-pinctrl" or "marvell,armada-3700-z1-sb-pinctrl"
 - reg: register specifier of MPP registers
 
 Available mpp pins/groups and functions:
@@ -12,7 +12,7 @@ Note: brackets (x) are not part of the mpp name for marvell,function and given
 only for more detailed description in this document. Customer should make sure
 that there are no mpps configuration conflicts.
 
-North bridge: "marvell,armada-3700-nb-pinctrl"
+North bridge: "marvell,armada-3700-z1-nb-pinctrl"
 name          pins     functions
 ================================================================================
 mpp0          0        i2c1(sck), gpio
@@ -53,7 +53,7 @@ mpp34         34       mmc, gpio
 mpp35         35       mmc, gpio
 mpp36         36       spi
 
-South bridge: "marvell,armada-3700-sb-pinctrl"
+South bridge: "marvell,armada-3700-z1-sb-pinctrl"
 name          pins     functions
 ================================================================================
 mpp0          0        usb32(drvvbus0), gpio
diff --git a/arch/arm64/boot/dts/marvell/armada-37xx-rz.dtsi b/arch/arm64/boot/dts/marvell/armada-37xx-rz.dtsi
index 8b4c98f..a45e9ed 100644
--- a/arch/arm64/boot/dts/marvell/armada-37xx-rz.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-37xx-rz.dtsi
@@ -48,4 +48,16 @@
 
 / {
 	model = "Marvell Armada 37xx RZ SoC";
+
+	soc {
+		internal-regs {
+			npinctrl: north-bridge-pinctrl@13830 {
+				compatible = "marvell,armada-3700-z1-nb-pinctrl";
+			};
+
+			spinctrl: south-bridge-pinctrl@18830 {
+				compatible = "marvell,armada-3700-z1-sb-pinctrl";
+			};
+		};
+	};
 };
diff --git a/drivers/pinctrl/mvebu/pinctrl-armada-3700.c b/drivers/pinctrl/mvebu/pinctrl-armada-3700.c
index 1eb013c..f665b1b 100644
--- a/drivers/pinctrl/mvebu/pinctrl-armada-3700.c
+++ b/drivers/pinctrl/mvebu/pinctrl-armada-3700.c
@@ -121,8 +121,8 @@ static int armada_3700_mpp_ctrl_set(unsigned pid,
 	return 0;
 }
 
-/* south bridge pin-ctl */
-static struct mvebu_mpp_mode armada_3700_nb_mpp_modes[] = {
+/* North bridge pin-ctl for Z1 */
+static struct mvebu_mpp_mode armada_3700_z1_nb_mpp_modes[] = {
 	MPP_MODE(0,
 	   MPP_FUNCTION(0x0, "i2c1", "sck"),
 	   MPP_FUNCTION(0x1, "gpio", NULL)),
@@ -246,8 +246,8 @@ static int armada_3700_mpp_ctrl_set(unsigned pid,
 	   MPP_FUNCTION(0x0, "spi", NULL)),
 };
 
-/* North bridge pins' configs setting bitmaps, this array idx is north bridge pin id. */
-static struct armada_3700_mpp_setting_bitmap armada_3700_nb_mpp_bitmap[] = {
+/* North bridge pins' configs setting bitmaps for Z1, this array idx is north bridge pin id. */
+static struct armada_3700_mpp_setting_bitmap armada_3700_z1_nb_mpp_bitmap[] = {
 	/* mask			config_num	configs */
 	{BIT(10),		2,		{0, BIT(10)}			},	/* pin 0 */
 	{BIT(10),		2,		{0, BIT(10)}			},	/* pin 1 */
@@ -335,8 +335,8 @@ static int armada_3700_nb_mpp_ctrl_set(unsigned pid, unsigned long config)
 	MPP_GPIO_RANGE(0, 0, 0, 36),
 };
 
-/* south bridge pin-ctl */
-static struct mvebu_mpp_mode armada_3700_sb_mpp_modes[] = {
+/* south bridge pin-ctl for Z1 */
+static struct mvebu_mpp_mode armada_3700_z1_sb_mpp_modes[] = {
 	MPP_MODE(0,
 	   MPP_FUNCTION(0x0, "usb32", "drvvbus0"),
 	   MPP_FUNCTION(0x1, "gpio", NULL)),
@@ -430,8 +430,8 @@ static int armada_3700_nb_mpp_ctrl_set(unsigned pid, unsigned long config)
 	   MPP_FUNCTION(0x1, "gpio", NULL)),
 };
 
-/* south bridge pins' configs setting bitmaps, this array idx is south bridge pin id */
-static struct armada_3700_mpp_setting_bitmap armada_3700_sb_mpp_bitmap[] = {
+/* south bridge pins' configs setting bitmaps for Z1, this array idx is south bridge pin id */
+static struct armada_3700_mpp_setting_bitmap armada_3700_z1_sb_mpp_bitmap[] = {
 	/*mask			config_num	configs*/
 	{BIT(0),		2,		{0, BIT(0)}			},	/* pin 0 */
 	{BIT(1),		2,		{0, BIT(1)}			},	/* pin 1 */
@@ -545,51 +545,51 @@ static unsigned int armada_3700_mpp_consistency_check(struct armada_3700_mpp_con
 static struct pinctrl_gpio_range armada_3700_sb_mpp_gpio_ranges[] = {
 	MPP_GPIO_RANGE(0, 0, 36, 30),
 };
-static struct mvebu_pinctrl_soc_info a3700_mpp_sb_soc_info = {
+static struct mvebu_pinctrl_soc_info a3700_z1_mpp_sb_soc_info = {
 	.variant = 0,
 	.controls = armada_3700_sb_mpp_controls,
 	.ncontrols = ARRAY_SIZE(armada_3700_sb_mpp_controls),
-	.modes = armada_3700_sb_mpp_modes,
-	.nmodes = ARRAY_SIZE(armada_3700_sb_mpp_modes),
+	.modes = armada_3700_z1_sb_mpp_modes,
+	.nmodes = ARRAY_SIZE(armada_3700_z1_sb_mpp_modes),
 	.gpioranges = armada_3700_sb_mpp_gpio_ranges,
 	.ngpioranges = ARRAY_SIZE(armada_3700_sb_mpp_gpio_ranges),
 };
 
-static struct armada_3700_mpp_conf a3700_mpp_sb_conf = {
-	.soc_info = &a3700_mpp_sb_soc_info,
-	.modes = armada_3700_sb_mpp_modes,
-	.nmodes = ARRAY_SIZE(armada_3700_sb_mpp_modes),
-	.bitmap = armada_3700_sb_mpp_bitmap,
-	.nbitmaps = ARRAY_SIZE(armada_3700_sb_mpp_bitmap),
+static struct armada_3700_mpp_conf a3700_z1_mpp_sb_conf = {
+	.soc_info = &a3700_z1_mpp_sb_soc_info,
+	.modes = armada_3700_z1_sb_mpp_modes,
+	.nmodes = ARRAY_SIZE(armada_3700_z1_sb_mpp_modes),
+	.bitmap = armada_3700_z1_sb_mpp_bitmap,
+	.nbitmaps = ARRAY_SIZE(armada_3700_z1_sb_mpp_bitmap),
 	.index = I_SOUTHBRIDGE,
 };
-static struct mvebu_pinctrl_soc_info a3700_mpp_nb_soc_info = {
+static struct mvebu_pinctrl_soc_info a3700_z1_mpp_nb_soc_info = {
 	.variant = 0,
 	.controls = armada_3700_nb_mpp_controls,
 	.ncontrols = ARRAY_SIZE(armada_3700_nb_mpp_controls),
-	.modes = armada_3700_nb_mpp_modes,
-	.nmodes = ARRAY_SIZE(armada_3700_nb_mpp_modes),
+	.modes = armada_3700_z1_nb_mpp_modes,
+	.nmodes = ARRAY_SIZE(armada_3700_z1_nb_mpp_modes),
 	.gpioranges = armada_3700_nb_mpp_gpio_ranges,
 	.ngpioranges = ARRAY_SIZE(armada_3700_nb_mpp_gpio_ranges),
 };
 
-static struct armada_3700_mpp_conf a3700_mpp_nb_conf = {
-	.soc_info = &a3700_mpp_nb_soc_info,
-	.modes = armada_3700_nb_mpp_modes,
-	.nmodes = ARRAY_SIZE(armada_3700_nb_mpp_modes),
-	.bitmap = armada_3700_nb_mpp_bitmap,
-	.nbitmaps = ARRAY_SIZE(armada_3700_nb_mpp_bitmap),
+static struct armada_3700_mpp_conf a3700_z1_mpp_nb_conf = {
+	.soc_info = &a3700_z1_mpp_nb_soc_info,
+	.modes = armada_3700_z1_nb_mpp_modes,
+	.nmodes = ARRAY_SIZE(armada_3700_z1_nb_mpp_modes),
+	.bitmap = armada_3700_z1_nb_mpp_bitmap,
+	.nbitmaps = ARRAY_SIZE(armada_3700_z1_nb_mpp_bitmap),
 	.index = I_NORTHBRIDGE,
 };
 
 static const struct of_device_id armada_3700_pinctrl_of_match[] = {
 	{
-		.compatible = "marvell,armada-3700-nb-pinctrl",
-		.data       = (void *) &a3700_mpp_nb_conf
+		.compatible = "marvell,armada-3700-z1-nb-pinctrl",
+		.data       = (void *) &a3700_z1_mpp_nb_conf
 	},
 	{
-		.compatible = "marvell,armada-3700-sb-pinctrl",
-		.data       = (void *) &a3700_mpp_sb_conf,
+		.compatible = "marvell,armada-3700-z1-sb-pinctrl",
+		.data       = (void *) &a3700_z1_mpp_sb_conf,
 	},
 	{ },
 };
-- 
1.7.9.5

